core/main/src/broker/endpoint_broker.rs

```diff
@@ -458,6 +458,10 @@ impl EndpointBrokerState {
 
     pub fn build_other_endpoints(&mut self, session: Option<AccountSession>) {
         for (key, endpoint) in self.rule_engine.rules.endpoints.clone() {
+            // skip thunder endpoint as it is already built using build_thunder_endpoint
+            if key == "thunder" {
```

Comment:
Maybe this should instead be:

`if let RuleEndpointProtocol::Thunder = endpoint.protocol {`

Reply:
Agreed.  Modified the code as suggested.